tests/many_to_one_null/tests.py

```diff
@@ -144,7 +144,11 @@ def test_related_null_to_field(self):
         c1 = Car.objects.create()
         d1 = Driver.objects.create()
         self.assertIs(d1.car, None)
-        with self.assertNumQueries(0):
+        msg = (
+            '"<Car: Car object (1)>" needs to have a value for field "make" before '
+            "this relationship can be used."
+        )
+        with self.assertRaisesMessage(ValueError, msg):
```

Comment:
This was something that we wanted to avoid (see [comment](https://code.djangoproject.com/ticket/19580#comment:49)). `c1` exists in the database, it doesn't have `make` so the list of drivers should be empty. We should probably copy `if self.instance.pk is None:` check to the `get_queryset()` instead of calling `_check_fk_val()` and raising an error for `foreign_related_fields`.